airbyte-ci/connectors_ci/metadata_service/orchestrator/orchestrator/sensors/catalog_sensors.py

```diff
@@ -0,0 +1,45 @@
+from typing import List
+from dagster import sensor, RunRequest, SkipReason, SensorDefinition, SensorEvaluationContext, build_resources, DefaultSensorStatus
+
+
+def generate_composite_etag_cursor(etags: List[str]):
+    return ":".join(etags)
+
+
+def catalog_updated_sensor(job, resources_def) -> SensorDefinition:
```

Comment:
```suggestion
def catalog_updated_sensor(job) -> SensorDefinition:
```

Don't you think it should be the sensor role to define which resources it needs and maybe create this?
If I'm following your implementation correctly we're defining global resources in __init__.py and we pass all these resources to this function. What if the `resources` dict grows with resources this sensor does not need? It will still "build_resources" on resources it does not need right?

Reply:
Thats a fair critique! If we add more resources we will end up building more than we needed to.

The only reason I did it this way is I didnt want new developers to add new resources to one of our reports and accidentally (and silently) break the sensor.

hmm let me investigate if theres a way we can solve both issues